mm2src/common/number_type_casting.rs

```diff
@@ -0,0 +1,124 @@
+/// This module contains type casting functions for numbers in a
+/// safe way.
+///
+/// The problem comes with when casting a type that supports higher value
+/// than the target type's highest value.
+///
+/// eg:
+/// ```rs
+/// let x: u64 = 4294967295 + 10;
+/// assert_eq!(x as u32, std::u32::MAX, "{} is not {}", x as u32, std::u32::MAX);
+/// ```
+
+pub trait SafeTypeCastingNumbers<T>: Sized {
+    fn into_or(self, or: T) -> T;
+    fn into_or_max(self) -> T;
+}
+
+macro_rules! impl_safe_number_type_cast {
+    ($from: ident, $to: ident) => {
+        impl SafeTypeCastingNumbers<$to> for $from {
+            fn into_or(self, or: $to) -> $to { std::convert::TryFrom::try_from(self).unwrap_or(or) }
+            fn into_or_max(self) -> $to { std::convert::TryFrom::try_from(self).unwrap_or(std::$to::MAX) }
+        }
+    };
+}
+
+// USIZE
+impl_safe_number_type_cast!(usize, isize);
+impl_safe_number_type_cast!(usize, i128);
+impl_safe_number_type_cast!(usize, i64);
+impl_safe_number_type_cast!(usize, i32);
+impl_safe_number_type_cast!(usize, i16);
+impl_safe_number_type_cast!(usize, i8);
+
+impl_safe_number_type_cast!(usize, u128);
+impl_safe_number_type_cast!(usize, u64);
+impl_safe_number_type_cast!(usize, u32);
+impl_safe_number_type_cast!(usize, u16);
+impl_safe_number_type_cast!(usize, u8);
+
+// ISIZE
+impl_safe_number_type_cast!(isize, i128);
+impl_safe_number_type_cast!(isize, i64);
```

Comment:
Good point, we should also cover signed integar to unsigned integar castings. But we can't rename the current `SafeTypeCastingNumbers` to `SafeTypeCastingUint` because it also provides signed-to-unsigned castings. Like casting `i64` to `u8`.

I think we can simply add `fn into_or_zero`  in `SafeTypeCastingNumbers`. What do you think ?

Reply:
I agree with Artem https://github.com/KomodoPlatform/atomicDEX-API/pull/1517#discussion_r1009387204
`into_or_zero` doesn't make sense to cast signed to unsigned as we should handle negative values manually